edweiss-firebase/functions/src/functions/lectures/addAudioTranscript.ts

```diff
@@ -24,19 +23,21 @@ type Lecture = LectureDisplay.Lecture;
 // -----------  Add Audio Transcript Cloud Function   ---------
 // ------------------------------------------------------------
 
-export const addAudioTranscript = onSanitizedCall(Functions.addAudioTranscript, {
-	courseId: Predicate.isNonEmptyString,
-	lectureId: Predicate.isNonEmptyString,
-	transcription: Predicate.isNonEmptyString,
-	pageNumber: Predicate.isStrictlyPositive,
-}, async (userId, args) => {
+export const addAudioTranscript = onAuthentifiedCall(Functions.addAudioTranscript, async (userId, args) => {
+	if (!args.courseId || !args.lectureId || !args.pageNumber || !args.transcription) {
+		return fail('invalid_arg');
+	}
+
 	const course = await getRequiredDocument(CollectionOf<Course>(`courses`), args.courseId, fail("course_not_found"));
+
 	const [lecture, lectureRef] = await getDocumentAndRef(CollectionOf<Lecture>(`courses/${args.courseId}/lectures`), args.lectureId);
 
 	if (lecture == undefined)
-		return fail("error_firebase");
+		return fail("firebase_error");
```

Comment:
why have you brought it back to firebase_error? In the model of LectureDoc you have defined that the possible error return values of addAudioTranscript to be error_firebase. It is better to follow this so as to maintain consistency

Reply:
If you prefer the format firebase_error then you could also replace it with error_firebase in the model if it bothers you that much

Reply:
Yes you are entirely right!

As you may have remembered, there was a bug with the audio transcript that way not showing off in Firebase. I had to use the wonderful command `git history` and `git blame` to understand what happened. I roll back some critical part to my earlier functionning version, which used ... `firebase_error` 🫣. My bad, I will correct it right away.

In aparte, I would like to stress that it's difficult to catch those bugs, as the code is executed not locally but on Firestore. I really had a pain to make it work again.

Reply:
Yes good job fixing it, and leave catching these small details of inconsistency to us reviewing you code